app/models/optout/OptOutContextData.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.optout
+
+import models.itsaStatus.ITSAStatus
+import models.itsaStatus.ITSAStatus.ITSAStatus
+import play.api.libs.json.{Json, OFormat}
+
+case class OptOutContextData(crystallisationStatus: Boolean,
+                             previousYearITSAStatus: String,
+                             currentYearITSAStatus: String,
+                             nextYearITSAStatus: String)
+
+object OptOutContextData {
+  implicit val format: OFormat[OptOutContextData] = Json.format[OptOutContextData]
+
+  def statusToString(status: ITSAStatus): String =
+    status match {
+      case ITSAStatus.NoStatus  => "U"
+      case ITSAStatus.Voluntary => "V"
+      case ITSAStatus.Annual    => "A"
+      case ITSAStatus.Mandated  => "M"
+      // This will be validated earlier on in a future ticket
+      case _ => throw new RuntimeException("Unexpected status")
```

Comment:
I feel throwing exceptions isn't within the spirit of the coding style of this project, using Either is more appropriate in my view

Reply:
Yes, this is a pragmatic solution to the problem here, and we have a ticket in the sprint to do this better.

I'm still unsure whether an exception or Either is better here. If we have an Either, then it may affect a lot of the code we have and we'll have to make sure this error case will weave its way through all the layers and show the error page. My thoughts are leaning towards using an exception as we've been promised that we should never get any of the other states from the API, and the Opt-Out design does not account for other states. To me it's pretty similar to not getting a response back from the API. Anyway, I'd rather defer this decision for now and pick up with the other ticket.